internal/cli/opsmanager/backup/config/update.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2020 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"github.com/mongodb/mongocli/internal/cli"
+	"github.com/mongodb/mongocli/internal/config"
+	"github.com/mongodb/mongocli/internal/flag"
+	"github.com/mongodb/mongocli/internal/store"
+	"github.com/mongodb/mongocli/internal/usage"
+	"github.com/spf13/cobra"
+	"go.mongodb.org/ops-manager/opsmngr"
+)
+
+const updateTemplate = "Backup Configuration updated.\n"
+
+type UpdateOpts struct {
+	cli.OutputOpts
+	cli.GlobalOpts
+	store             store.BackupConfigUpdater
+	clusterID         string
+	status            string
+	storageEngine     string
+	authMechanism     string
+	username          string
+	password          string
+	encryption        bool
+	ssl               bool
+	syncSource        string
+	provisioned       bool
+	excludedNamespace []string
+	includedNamespace []string
+}
+
+func (opts *UpdateOpts) initStore() error {
+	var err error
+	opts.store, err = store.New(config.Default())
+	return err
+}
+
+func (opts *UpdateOpts) Run() error {
+	r, err := opts.store.UpdateBackupConfig(opts.newBackupConfig())
+
+	if err != nil {
+		return err
+	}
+
+	return opts.Print(r)
+}
+
+func (opts *UpdateOpts) newBackupConfig() *opsmngr.BackupConfig {
+	return &opsmngr.BackupConfig{
+		GroupID:            opts.ConfigProjectID(),
+		ClusterID:          opts.clusterID,
+		StatusName:         opts.status,
+		StorageEngineName:  opts.storageEngine,
+		EncryptionEnabled:  opts.encryption,
+		SSLEnabled:         opts.ssl,
```

Comment:
This is along with Provisioned and EncryptionEnabled are a bit dangerous, if the user doesn't set the flag then it will disable  `SSLEnabled` or `Provisioned`, we should use those flags only to enable we can look later to add new flags to disable them

Reply:
this also brings back the point where these fields should be pointers and omitempty